pkg/catalog/types.go

```diff
@@ -12,6 +12,42 @@ type ComponentType struct {
 	Spec              ComponentSpec `json:"spec,omitempty"`
 }
 
+// DevfileComponentType is the main struct for devfile catalog components
+type DevfileComponentType struct {
+	Name        string
+	DisplayName string
+	Description string
+	Link        string
+	Support     bool
+}
+
+// DevfileIndexEntry is the main struct of index.json from devfile registry
+type DevfileIndexEntry struct {
+	DisplayName       string   `json:"displayName"`
+	Description       string   `json:"description"`
+	Tags              []string `json:"tags"`
+	Icon              string   `json:"icon"`
+	GlobalMemoryLimit string   `json:"globalMemoryLimit"`
+	Links             struct {
+		Link string `json:"self"`
+	} `json:"links"`
+}
+
+// Devfile is the main structure of devfile from devfile registry
+type Devfile struct {
+	APIVersion string `yaml:"apiVersion"`
+	MetaData   struct {
+		GenerateName string `yaml:"generateName"`
+	} `yaml:"metadata"`
+	Components []struct {
+		Type  string `yaml:"type"`
+		Alias string `yaml:"alias"`
+	} `yaml:"components"`
+	Commands []struct {
+		Name string `yaml:"name"`
+	} `yaml:"commands"`
+}
```

Comment:
See other comment, this way of implementing nested structs is unconventional

Reply:
Actually nested struct is not a typical way to frequently use for general struct, but it's a very convenient way for defining format for `json.Unmarshal()` and `yaml.Unmarshal()` because we can easily check the whole format of the `json` or `yaml` file instead of looking up the mapping relationship between structs.

Topics regarding `Nested structs in Golang`:
Source from Medium: https://medium.com/@xcoulon/nested-structs-in-golang-2c750403a007
Source from Go Playground: https://play.golang.org/p/4rGB5da0Lt